domain/src/com/thoughtworks/go/domain/BuildCommand.java

```diff
@@ -144,7 +144,9 @@ public static BuildCommand error(String message) {
     @Expose
     private BuildCommand onCancel;
     @Expose
-    private String[] execInput;
+    private String[] execInput = new String[0];
+    @Expose
+    private Map<String, String> commandEnvVars = new HashMap<String, String>();
```

Comment:
At some point later, I might move both of these out to the ExecCommand's inputs, because this feels weird to me. A BuildCommand can be anything, including "test" and "reportCompletion". Adding execInput and commandEnvVars here seems wrong, because they are both specific to ExecCommand only. Am I making sense?